### PR TITLE
Casting message to bytes in PyCrypto verifier.

### LINKED ISSUE #201: Python 3 + PyCrypto: crypt.verify_signed_jwt_with_certs raises "Invalid token signature" due to string/bytes bug
I was getting a mysterious "Invalid token signature" error trying to validate Google Sign-In-generated tokens using this method on Python 3.4.0 with PyCrypto. After confirming with jwt.io and the `tokeninfo` endpoint that the token was, indeed, valid, I dove into the method source. [After realizing that all exceptions were swallowed in the PyCrypto wrapper's `verify` method](https://github.com/google/oauth2client/blob/c5eb9f14d46d89fb33c4409a0d32fad4b919e136/oauth2client/crypt.py#L231-L235), I found out that the call:

```python
SHA256.new(message)
```

was raising:

```python
TypeError: Unicode-objects must be encoded before hashing
```

This is because `message` is a string, and not bytes. This isn't surprising when you see where `message` is initially set: 
https://github.com/google/oauth2client/blob/c5eb9f14d46d89fb33c4409a0d32fad4b919e136/oauth2client/crypt.py#L413.

I applied a quick fix to that line:

```python
signed = ('%s.%s' % (segments[0], segments[1])).encode('utf-8')
```

And the method validated my token correctly.

In addition to this fix, I'd recommend removing that `try/except` block - I'm not sure why you'd want to swallow exceptions there, other than to make it harder for poor developers like me to debug issues ;)

### PR DESCRIPTION
**Pull request by dhermes**
Also
- Removing bare `except:` statements in both the PyCrypto and
  OpenSSL verifiers (in the `verify` method).
- Catching the only exception possible in the OpenSSL verifier
  (it is `OpenSSL.crypto.Error`).
- Converting the signature to bytes (if not already) in the
  OpenSSL verifier.
- Adding a test with both a unicode and bytes signature for
  each verifier.

Fixes #201.

/cc @nathanielmanistaatgoogle

### CODE REVIEW

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle @craigcitro As mentioned above, I removed the bare `except:` statements the verifiers. It's unclear if they were there intentionally or just done since the exception types were neglected.

File: tests/test_jwt.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def test_sign_and_verify(self):
     self._check_sign_and_verify('privatekey.%s' % self.format)
 
   def test_sign_and_verify_from_converted_pkcs12(self):
-    """Tests that following instructions to convert from PKCS12 to PEM works."""
```
Comment (nathanielmanistaatgoogle) [inline tests/test_jwt.py:62]: What's up with the change on this line?

File: tests/test_jwt.py [hunk 2]
```diff
@@ -196,6 +198,7 @@ def setUp(self):
 
 
 class PEMCryptTestsOpenSSL(CryptTests):
+
```
Comment (nathanielmanistaatgoogle) [inline tests/test_jwt.py:201]: Any comment on these blank line insertions?

File: oauth2client/crypt.py [hunk 3]
```diff
@@ -93,12 +93,14 @@ def verify(self, message, signature):
         key that this object was constructed with.
       """
       from OpenSSL import crypto
+      if isinstance(message, six.text_type):
```
Comment (nathanielmanistaatgoogle) [inline oauth2client/crypt.py:98]: Any need to update the doc string for these arguments from "string"?

File: tests/test_jwt.py (hunk 1 above)
Comment (dhermes) [inline tests/test_jwt.py:62]: Docstrings on tests break the output of `nosetest` (and are generally frowned upon).

I can leave it out of this commit, I just noticed it because it showed up in a Py3.4 failure I fixed before adding

``` python
      if isinstance(signature, six.text_type):
        signature = signature.encode('utf-8')
```

to `OpenSSLVerifier.verify`.

Comment (nathanielmanistaatgoogle) [inline tests/test_jwt.py:62]: Wait, I've without knowing been doing something wrong for years _again_? Can you link me to a source for this general frowning?

File: tests/test_jwt.py (hunk 2 above)
Comment (dhermes) [inline tests/test_jwt.py:201]: Just consistency with `CryptTests`. I thought the extra line was a requirement for the linter, but after some experimenting it doesn't seem to be.

File: oauth2client/crypt.py (hunk 3 above)
Comment (dhermes) [inline oauth2client/crypt.py:98]: Yeah happy to do it.

File: tests/test_jwt.py (hunk 1 above)
Comment (dhermes) [inline tests/test_jwt.py:62]: Couldn't find anything in a few minutes of searching. I'm pretty sure I've seen something before.

Closest I got was the `unittest` docs for making the description ignore the docstring:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.shortDescription

Maybe @tseaver has a quick reference

Comment (tseaver) [inline tests/test_jwt.py:62]: Docstrings in tnittest-based testcase methods are a bad idea because (historically, anyway) testrunners tend to display the docstring, rather than the name of the testcase, in output, making it hard to diagnose failures.

**Comment by craigcitro** [comment]
i don't know that there's a reason for the bare `except:` clauses -- but if there is, we'll probably get a bug report soon enough. :wink:

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle I have updated the docstrings on the updated `verify()` methods.

Sorry I let this thing fall away.

Are we good to go here?

**Comment by nathanielmanistaatgoogle** [comment]
LGTM; I think we're good to go here.

**Comment by dhermes** [comment]
Thanks!
